### PR TITLE
A11y

### PR DESCRIPTION
**Pull request by oxyc**
There are at least two bugs.

1. ~Not sure where the styling for the submenu went?~

2. ~This styling is not ideal but it's quite complicated to get it correct....~

There's one annoying breaking change here: https://github.com/generoi/genero-design-system/commit/faf842434449adb3728bc780577a72384dd31cfb

The above commit defaults menus to align left. We could also default it to align-center, then the `<gds-navigation>` component would look correct but the `<gds-menu>` used directly would be centered instead of aligned left. This is less of a breaking change but I think it's a weird default.

This should make all the components tabbable correctly and have basic accessibility but it still needs real life testing. The menu could also use keyboard navigation beyond tabbing but I don't think that's a strict requirement since it does work through tabbing.

### CODE REVIEW

**Comment by oxyc** [comment]
I found it difficult to get the proper focus box sizes when there were quite many wrappers around everything. So in commit https://github.com/generoi/genero-design-system/pull/41/commits/4c7edc9540c891b7cf4fc4aa25abd35a2ff2e5a6 the focus styling on menu-item-nested looks good and most components got simplified.
 
This is quite opinionated though, I personally prefer having fewer wrappers around. What do others think? I can easily drop this commit and we can look at the focus style boxes later since it's not very important.
 
Here are the relevant diffs of the DOM change:
https://github.com/generoi/genero-design-system/pull/41/commits/4c7edc9540c891b7cf4fc4aa25abd35a2ff2e5a6#diff-639cc5987fa2b5be60df8950700bb506e781a89d6f0b0d84170094216e7f8c28L98-R118
 
https://github.com/generoi/genero-design-system/pull/41/commits/4c7edc9540c891b7cf4fc4aa25abd35a2ff2e5a6#diff-569ecd2879518dceeefe2e932603fb2984fc28a0e2034c086a5fbf133a0a0360L36-R38
 
https://github.com/generoi/genero-design-system/pull/41/commits/4c7edc9540c891b7cf4fc4aa25abd35a2ff2e5a6#diff-7b5bcf3c223e6f4f72ca8f9220883641981b01e0beaef0a2374784c14156f175L37-L44
 
https://github.com/generoi/genero-design-system/pull/41/commits/4c7edc9540c891b7cf4fc4aa25abd35a2ff2e5a6#diff-260768a9e14dfcfd429f0bfbc3b9496ef4d62f63fa101c02e0cf437405665247L34-L38

File: src/components/gds-navigation/gds-navigation.stories.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ export const Examples = () => html`
       --icon-color-hover: var(--navigation-color);
       --navigation-content-background: repeating-linear-gradient(to right, #166e3b, #166e3b 10px, #146635 10px, #146635 20px);
       --menu-item-background-color-active: transparent;
-      --menu-item-underline-display: block;
+      --navigation-underline-active: underline solid currentColor 3px;
```
Comment (oxyc) [inline src/components/gds-navigation/gds-navigation.stories.js:99]: Need to rename it at least...

File: src/components/gds-menu-item-nested/gds-menu-item-nested.scss [hunk 2]
```diff
@@ -53,70 +24,95 @@ li {
...
+  padding: 0;
+
+  // Reset browser default button styles
+  appearance: none;
+  background: none;
+  border: 0;
+  outline: none;
+  font-size: inherit;
+  line-height: inherit;
+  margin-left: -1em;
```
Comment (oxyc) [inline src/components/gds-menu-item-nested/gds-menu-item-nested.scss:40]: Should use a variable so this makes sense

File: src/components/gds-menu-item-nested/gds-menu-item-nested.scss [hunk 3]
```diff
@@ -53,70 +24,95 @@ li {
...
+  @include desktopOnly {
+    padding-left: var(--spacing-s);
+    padding-right: var(--spacing-s);
+  }
+}
+
+// Focus link styles
+::slotted(a:focus) {
+  outline: none;
+  box-shadow: var(--menu-link-box-shadow-focus);
```
Comment (oxyc) [inline src/components/gds-menu-item-nested/gds-menu-item-nested.scss:116]: Variable name isn’t ideal

File: src/components/gds-menu/gds-menu.stories.js [hunk 4]
```diff
@@ -35,12 +35,12 @@ export const Examples = () => html`
 
 // prettier-ignore
 export const Submenus = () => html`
-  <gds-menu>
+  <gds-menu role="menubar" aria-label="Primary navigation">
```
Comment (oxyc) [inline src/components/gds-menu/gds-menu.stories.js:38]: Not sure if we should just use accessible-label for consistency even if the aria attribute works in this case

File: src/components/gds-submenu/gds-submenu.scss [hunk 5]
```diff
@@ -1,32 +1,35 @@
...
 ::slotted(a) {
   color: unset;
   text-decoration: none;
+  display: block;
+}
+
+// Focus link styles
+::slotted(a:focus) {
+  outline: none;
+  box-shadow: var(--menu-submenu-link-box-shadow-focus);
```
Comment (oxyc) [inline src/components/gds-submenu/gds-submenu.scss:34]: Variable name...

**Comment by oxyc** [comment]
Screen Shot 2021-03-28 at 14 49 33

Updated the styles to use outline, we'll live without support for border radius.

**Comment by oxyc** [comment]
## [4.0.0-beta.0](https://github.com/generoi/genero-design-system/compare/v3.13.1...v4.0.0-beta.0) (2021-04-02)


### ⚠ BREAKING CHANGES

*  **Menu item underline**

`--menu-item-underline-display` has been replaced by
`--gds-menu-item-underline-active` and the underlines are no longer spanning
the entire block, only the text content. Additionally you can modify the offset and thickness; and
`--navigation-underline-color` is now called `--gds-menu-item-underline-color-active`.

```scss
// BEFORE:
--menu-item-underline-display: none;
--navigation-underline-color: var(--color-black);

// AFTER:
--gds-menu-item-underline-active: underline;
--gds-menu-item-underline-thickness-active: 3px;
--gds-menu-item-underline-color-active: currentColor;
--gds-menu-item-underline-offset-active: var(--spacing-xxs);
```

**Menu variables renamed**

All `--menu-*` variables have been prefixed with `--gds-menu` instead.

```scss
// BEFORE:
--menu-text-align: left;
--menu-item-background-color-active: var(--background-color-primary);

// AFTER:
--gds-menu-text-align: left;
--gds-menu-item-background-color-active: var(--background-color-primary);
```

**Markup change for ´<gds-menu-nested-item>`**

Using `<gds-menu-nested-item>` should now wrap the link content in `<gds-menu-item>`.

```diff
 <gds-menu>
   <gds-menu-item-nested slot="item" submenu-icon="❯">
     <a slot="link" href="#first">
-      First item
+      <gds-menu-item>First item</gds-menu-item>
     </a>
   </gds-menu-item-nested>
 </gds-menu>
```

With the above change, the `--gds-menu-item-nested-a-padding` variable has now
been removed and the padding is instead styled from the `<gds-menu-item>`
component.

```scss
// REMOVED:
--gds-menu-item-nested-a-padding: var(--spacing-xs) 0;
```

**Navigation gutters**

Durings this refactoring the fixed menu height that used to be set
`var(--spacing-xxl)`, is now fluid and instead uses the following variables for
adjusting the padding on `<gds-menu-item>` depending on the direction of the
`<gds-menu>` component:

```scss
--gds-menu-item-padding-horizontal: var(--spacing-s) var(--spacing-s) calc(var(--spacing-s) - 0.15em) var(--spacing-s);
--gds-menu-item-padding-vertical: var(--spacing-s) var(--spacing-m) calc(var(--spacing-s) - 0.15em) var(--spacing-m);
```

This is likely to break some implementations or at least change the height of the navigation by a few pixels.

The navigation containers now supports `--navigation-padding` to consistently
set left/right padding. You most likely need to change some theme styles for
the right padding of the navigation as well as reduce the left padding for the
navigation logo `--navigation-logo-padding`. By default the
`--navigation-padding` sets the same values as before, making spacing identical
as it used to on both desktop and mobile. If you override the variable you
should consider the spacing both for desktop and for mobile.

```scss
// Before
--navigation-logo-padding: 0 0 0 var(--spacing-s);

// After
--navigation-logo-padding: 0;
```

**Navigation hamburger icon**

The navigations default mobile hamburger has now been changed to an animated
`<gds-hamburger>` component, if you applied color styling to this in your theme
you might need to make adjustments.

**Navigation menu item justification**

`<gds-menu>` used within a `<gds-navigation>` needs to specify
that the menu items should be centered to maintain the previous behavior
otherwise the menu will be aligned to the left.

Add this to your css:

```scss
// NEW VARIABLE:
--gds-menu-justify-content: center;
```

**`<gds-input-wrapper>` no longer using shadow-dom**

Due to accessibility issues and it not being possible to reference input fields
in a slot from a `<label>` inside a component, the `<gds-input-wrapper>` is now
using light-dom and scoped styles instead of shadow-dom. This could cause
high specificity global styles to pollute the component.

### Features

* **a11y:** make all components accessible ([b5b069c](https://github.com/generoi/genero-design-system/commit/b5b069c62b207ed1af69cef778bacb8540f14783))
* **gds-hamburger:** add component ([3a290c4](https://github.com/generoi/genero-design-system/commit/3a290c4d803724c0ce96456568ca44c3d6490d8e))
* **gds-input-wrapper:** add placeholder color variable ([51a265b](https://github.com/generoi/genero-design-system/commit/51a265b7820fb0f32ef97e19db5928dd7aa77ffe))
* **gds-input-wrapper:** style select inputs on safari ([68d7923](https://github.com/generoi/genero-design-system/commit/68d7923c3ac4e44796bc4a128ada5e01e8d440be))
* **gds-link:** add full mode ([b0c68f7](https://github.com/generoi/genero-design-system/commit/b0c68f7ca140038c2c8e03d8b8b3fbaf4ead20ce))
* **gds-navigation:** add more css variables ([9145b40](https://github.com/generoi/genero-design-system/commit/9145b40cf8d668d51d74e6698afec949fc507cb3))
* **gds-search-form:** add component ([f89aec0](https://github.com/generoi/genero-design-system/commit/f89aec0e8fc47dd18f2da22d0ac8d96950a2e984))
* **gds-visually-hidden:** add component ([e9bf15b](https://github.com/generoi/genero-design-system/commit/e9bf15bf21efe954ede47915ab1356ad3bf7af28))
* **storybook:** Add support for story controls ([bd51bf8](https://github.com/generoi/genero-design-system/commit/bd51bf82576725d7c747ce774791f69b7660c2d2))
* **submenu:** add submenu and related components ([9408cad](https://github.com/generoi/genero-design-system/commit/9408cad892d5400262baf81fe4a757de3d0ce7b3))
* **submenu:** add support for mobile submenu ([544e8bd](https://github.com/generoi/genero-design-system/commit/544e8bd0f1303d2abc666cce59e383c376647b10))


### Bug Fixes

* **audit:** update dependencies ([fe9b644](https://github.com/generoi/genero-design-system/commit/fe9b6440e780e5e7f5cc1e39e35c816bf263075a))
* **gds-hint:** fix overflow in safari ([36bc0b7](https://github.com/generoi/genero-design-system/commit/36bc0b77e201ef61b740e7fa55ce0ed5e36b1d27))
* **safari:** use all:unset to reset button and input styling ([0671da8](https://github.com/generoi/genero-design-system/commit/0671da8d9115ee1c951b04b884a5cf28c3c05bd8))

**Comment by taromorimoto** [comment]
@oxyc Great release notes! ❤️ 

Is there a reason you are omitting the `--gds` prefixing `variable-names` branch?

**Pull request merged**